Jenkinsfile

```
@@ -27,37 +27,40 @@ if (needSplittingFromWorkspace) {
-                                // The build failure prevents parallel tests executor to realize the tests are gone so same
-                                // split is run to execute and report zero tests - which fails the build. Permit the test
-                                // results to be empty to break the circle: build after removal executes one empty split
-                                // but not letting the build to fail will cause next build not to try those tests again.
-                                allowEmptyResults: true
-                        ) {
-                            sh """
-                                set-java.sh $javaVersion
-                                eval \$(vnc.sh)
-                                java -version
+      for (String v in ["lts", "latest"]) {
+          int javaVersion = j
+          String jenkinsUnderTest = v
+          def name = "java-${javaVersion}-jenkins-${jenkinsUnderTest}split${index}"
+          branches[name] = {
+              stage(name) {
+                  node('docker-highmem') {
+                      checkout scm
+                      def image = docker.build('jenkins/ath', '--build-arg uid="$(id -u)" --build-arg gid="$(id -g)" ./src/main/resources/ath-container/')
+                      image.inside('-v /var/run/docker.sock:/var/run/docker.sock --shm-size 2g') {
+                          def exclusions = splits.get(index).join("\n")
+                          writeFile file: 'excludes.txt', text: exclusions
+                          realtimeJUnit(
+                                  testResults: 'target/surefire-reports/TEST-*.xml',
+                                  testDataPublishers: [[$class: 'AttachmentPublisher']],
+                                  // Slow test(s) removal can causes a split to get empty which otherwise fails the build.
+                                  // The build failure prevents parallel tests executor to realize the tests are gone so same
+                                  // split is run to execute and report zero tests - which fails the build. Permit the test
+                                  // results to be empty to break the circle: build after removal executes one empty split
+                                  // but not letting the build to fail will cause next build not to try those tests again.
+                                  allowEmptyResults: true
+                          ) {
+                              sh """
+                                  set-java.sh $javaVersion
+                                  eval \$(vnc.sh)
+                                  java -version
 
-                                run.sh firefox latest -Dmaven.test.failure.ignore=true -DforkCount=1 -B
-                            """
-                        }
-                    }
-                }
-            }
+                                  run.sh firefox jenkinsUnderTest -Dmaven.test.failure.ignore=true -DforkCount=1 -B
```

Does the `run.sh` script expand jenkinsUnderTest to be the correct value or is there something else happening here that somehow expands that String variable?
```suggestion
                                  run.sh firefox ${jenkinsUnderTest} -Dmaven.test.failure.ignore=true -DforkCount=1 -B
```

---

😆  nice catch

---

no I just messed up doing a quick patch in `notepad2`